.github/workflows/main.yaml

```diff
@@ -43,33 +76,174 @@ jobs:
       - name: Create runner
         run: |
           gcloud compute instances create ${{ steps.generator.outputs.runner }} \
-            --zone ${{ vars.GCP_ZONE }} \
+            --zone ${{ inputs.zone }} \
             --source-instance-template ${{ inputs.runner_template }}
       - name: Create PAT token secret
         run: |
           echo -n ${{ secrets.PAT_TOKEN }} \
             | gcloud secrets create PAT_TOKEN_${{ steps.generator.outputs.uuid }} --data-file=-
+
+  installation:
+    if: ${{ always() && needs.create-runner.result == 'success' }}
+    runs-on: ${{ needs.create-runner.outputs.uuid }}
+    needs:
+      - create-runner
+    outputs:
+      PUBLIC_IP: ${{ steps.runner-ip.outputs.PUBLIC_IP }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+            
+      - name: Get Runner IP
+        id: runner-ip
+        run: echo "PUBLIC_IP=$(curl -s ifconfig.co)" >> "$GITHUB_OUTPUT"
+            
+      - name: Install K3s / Helm / Rancher
+        env:
+          KUBECONFIG: /etc/rancher/k3s/k3s.yaml
+          HELM_VERSION: 3.13.1
+          K3S_VERSION: v1.26.10+k3s1
```

Comment:
I think K3S_VERSION should also be added to the `input` list since we are bound to test on upgraded k8s version every few months. WDYT?

Reply:
I too thought that initially, added now as input